### PR TITLE
Add implementation of PRESTO middleware.

### PR DESCRIPTION
**Pull request by cheatfate**

### CODE REVIEW

File: presto/middleware.nim [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#         Status Research & Development GmbH
+#
+#              Licensed under either of
+#  Apache License, version 2.0, (LICENSE-APACHEv2)
+#              MIT license (LICENSE-MIT)
+
+{.push raises: [].}
+
+import chronos, chronos/apps/http/httpserver
+import route, servercommon, serverprivate
```
Comment (arnetheduck) [inline presto/middleware.nim:13]: `"."/[route, ...]` avoids the risk of loading these modules from a different package

Comment (cheatfate) [reply presto/middleware.nim]: Fixed in https://github.com/status-im/nim-presto/pull/73/commits/0fc1d8dbdec54b75deff52789042ec63f3ceaf3d

File: presto/serverprivate.nim [hunk 2]
```diff
@@ -107,230 +107,255 @@ proc processRestRequest*[T](
...
 
       when defined(metrics):
-        presto_server_invalid_requests_count.inc()
+        processStatusMetrics(route, Http410)
+
+      await request.respond(Http410)
+
+    of RestApiResponseKind.Status:
+      var headers = HttpTable.init()
+      if server.router.allowedOrigin.isSome:
```
Comment (NagyZoltanPeter) [inline presto/serverprivate.nim:246]: Forgive me to put my request here as its easier to point out.
This is the place which I would like to have it decoupled from the core processing of rest requests.
As it is ok that I can chain rest request processors as middlewares but having allow-origin handled here I will not be (or just with unnecessage response header hacking) to separately influence this logic.
It woulbe beneficial if I can use the rest middleware where I can put my CORS logic separately, ideally in a separate middleware and injected into the chain.
WDYT is it solveable?

Comment (cheatfate) [reply presto/serverprivate.nim:242]: You can disable internal PRESTO CORS mechanism if you create `RestRouter` instance with supplying `allowedOrigin` argument.

**Review by NagyZoltanPeter — APPROVED** [review]
LGTM!
I have finished (after couple of downprio) CORS implementation of nim-waku using this PR. It is working.
I will need to PR to be merged and released to finalize my PR on nwaku.

Thank you.

**Pull request merged**